src/gz/files.c

```diff
@@ -460,6 +464,22 @@ static void scroll_down_proc(struct menu_item *item, void *data)
     ds->scroll = 0;
 }
 
+static int mkdir_osk_callback_proc(const char *str, void *data)
+{
+    char path_buf[256];
+    getcwd(path_buf, sizeof(path_buf));
+    strncat(path_buf, "/", sizeof(path_buf) - 1);
+    strncat(path_buf, str, sizeof(path_buf) - 1);
+    mkdir(path_buf, S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);
```

Comment:
`mkdir` doesn't need an absolute path, you can just pass `str` into it, and it will make the new directory in the current directory. The default permissions for new directories is usually `0755`, or `S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH`. Ideally you would also check the return value and display a message if there is an error.

Reply:
From discussion on discord:
Removing line 471 results in sub directories not being able to be created, possibly a bug in mkdir.